### ISSUE TITLE
Fast import crash

### ISSUE THREAD

**Issue by sveneld** [newren/git-filter-repo#88]
Command
`git filter-repo --path protected/modules/ --invert-paths`

give me an error
```
Parsed 64782 commitsfatal: Invalid raw date "4559547106 -7349423" in ident: Alexandr <alexandr@example.com> 4559547106 -7349423

fast-import: dumping crash report to .git/fast_import_crash_25675
```

How it can be fixed?

**Comment by sveneld** [comment]
I found information in your documentation
```
Issues specific to filter-repo
Multiple repositories in the wild have been observed which use a bogus timezone (+051800); google will find you some reports. The intended timezone wasn’t clear or wasn’t always the same. Replace with a different bogus timezone that fast-import will accept (+0261).
```

so i add into function _parse_user such code
```
if when.endswith(b'-7349423'):
      when = b'1403856436 +0300'
```

**Comment by newren** [comment]
Yeah, fast-import is somewhat strict on what it'll accept and this is clearly a bogus time.  Luckily, fast-export and filter-repo are both kind of lax about parsing this, which gives you a chance to edit it before fast-import tries to parse it.  Perhaps you could run with the following four liner:

```
git filter-repo --path protected/modules/ --invert-paths --commit-callback '
  if commit.author_date == b"4559547106 -7349423":
    commit.author_date = b"1403856436 +0300"
'
```

You may need to replace "author_date" with "committer_date", or check and update both.

Does that help?

**Comment by rjbell4** [comment]
FYI, I also just ran into this:
```
fatal: Invalid raw date "5859358690 -43455309" in ident: vmeng <vmeng@isilon.com> 5859358690 -43455309
fast-import: dumping crash report to fast_import_crash_27805
```

From the crash report (filtered in obvious ways for privacy):
```
  get-mark :1034676
  ls :998562 path/to/file.xml
  commit refs/pull/1824/head
  mark :1034677
  author user <user@domain.com> 1560842006 -0500
* committer user <user@domain.com> 5859358690 -43455309
```

Which Git pretty-prints as:
```
commit c4cd121122abdbf78b0b76a15c2089cb6ee09f9c (refs/pull/1824/head)
Author:     user <user@domain.com>
AuthorDate: Tue Jun 18 02:13:26 2019 -0500
Commit:     user <user@domain.com>
CommitDate: Sun Feb 7 06:29:10 2106 -43455309
```

I have no idea how that happened.

I am going to try using `--commit-callback` to have the `CommitDate` match the `AuthorDate`.

**Comment by rjbell4** [comment]
FYI, I also just ran into this:
```
fatal: Invalid raw date "5859358690 -43455309" in ident: user <user@domain.com> 5859358690 -43455309
fast-import: dumping crash report to fast_import_crash_27805
```

From the crash report (filtered in obvious ways for privacy):
```
  get-mark :1034676
  ls :998562 path/to/file.xml
  commit refs/pull/1824/head
  mark :1034677
  author user <user@domain.com> 1560842006 -0500
* committer user <user@domain.com> 5859358690 -43455309
```

Which Git pretty-prints as:
```
commit c4cd121122abdbf78b0b76a15c2089cb6ee09f9c (refs/pull/1824/head)
Author:     user <user@domain.com>
AuthorDate: Tue Jun 18 02:13:26 2019 -0500
Commit:     user <user@domain.com>
CommitDate: Sun Feb 7 06:29:10 2106 -43455309
```

I have no idea how that happened.

I am going to try using `--commit-callback` to have the `CommitDate` match the `AuthorDate`.

**Comment by newren** [comment]
Discussion on the mailing list was that GitHub relaxed their checks years ago to just allow these types of bogus dates to be pushed, and that we should just relax fast-import to do the same (actually the suggestion was to fix fast-export to normalize them, but that would prevent people from being able to add filter-repo callbacks to fix them up, so I'd rather that all went into fast-import); see the thread starting at https://lore.kernel.org/git/CABPp-BFfa6q96qMUN07Dq3di6d3WuUzhyktBytbX=FGgarXgjg@mail.gmail.com/

**Comment by newren** [comment]
https://lore.kernel.org/git/pull.795.v2.git.git.1590698437607.gitgitgadget@gmail.com/; that will require git-2.28.0 (I submitted a few weeks too late to make it into the upcoming git-2.27.0), and will require me to add some code detecting whether the user has that git version so that filter-repo can safely make use of that new flag.

**Comment by newren** [comment]
Made use of the new --anonymize-map option of fast-export in the fast-export -h output to key whether I could safely add this flag.  So, this is now fixed with commit cefeef1c0a (filter-repo: use new --date-format=raw-permissive fast-import option, 2020-06-06), assuming you are using a new enough (i.e. not-yet-released) version of git.
